### PR TITLE
feat: add ofetch

### PR DESCRIPTION
**Pull request by marclave**
a baseline for ofetch integration :)

### CODE REVIEW

File: packages/snippetz-plugin-js-ofetch/src/ofetch.test.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+          hello: 'world',
+        }),
+      },
+    })
+
+    expect(source.code).toBe(`ofetch('https://example.com', {
+  headers: {
+    'Content-Type': 'application/json'
+  },
+  body: JSON.stringify({
```
Comment (pi0) [inline packages/snippetz-plugin-js-ofetch/src/ofetch.test.ts:72]: We don't need to explicitly use `JSON.stringify` neither add `Content-Type` both will be auto added

File: packages/snippetz-plugin-js-ofetch/src/ofetch.test.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+          value: 'bar',
+        },
+        {
+          name: 'bar',
+          value: 'foo',
+        },
+      ],
+    })
+
+    expect(source.code).toBe(`ofetch('https://example.com?foo=bar&bar=foo')`)
```
Comment (pi0) [inline packages/snippetz-plugin-js-ofetch/src/ofetch.test.ts:93]: You can generate with `ofetch(url, { query: { ... } }`

File: packages/snippetz-plugin-node-ofetch/src/ofetch.test.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+          hello: 'world',
+        }),
+      },
+    })
+
+    expect(source.code).toBe(`ofetch('https://example.com', {
+  headers: {
+    'Content-Type': 'application/json'
+  },
+  body: JSON.stringify({
```
Comment (pi0) [inline packages/snippetz-plugin-node-ofetch/src/ofetch.test.ts:72]: also here

File: packages/snippetz-plugin-js-ofetch/src/ofetch.test.ts (hunk 2 above)
Comment (marclave) [reply packages/snippetz-plugin-js-ofetch/src/ofetch.test.ts:93]: is this the preferred means? : ) cause we have control on how we want to showcase the snippets ✨

Comment (pi0) [reply packages/snippetz-plugin-js-ofetch/src/ofetch.test.ts:84]: Normally yes for consistency. Any reason you are worries to render `query` opt and prefer inline?

Comment (marclave) [reply packages/snippetz-plugin-js-ofetch/src/ofetch.test.ts:84]: no preference, haha was just habit! I prefer the query as field vs inline! let me fix that up ✨

Comment (marclave) [reply packages/snippetz-plugin-js-ofetch/src/ofetch.test.ts]: done :)

**Comment by pi0** [comment]
Thinking about this, are you going to render `import { ofetch } from 'ofetch'` in instructions?

While we document `ofetch` in docs, you could alternatively render with `$fetch` which works out of the box with Nitro, Nuxt and other Nitro-based backends.

**Comment by marclave** [comment]
> Thinking about this, are you going to render `import { ofetch } from 'ofetch'` in instructions?
> 
> While we document `ofetch` in docs, you could alternatively render with `$fetch` which works out of the box with Nitro, Nuxt and other Nitro-based backends.

yes it would be `import { ofetch } from 'ofetch'`, but what we can do is add `Nuxt` `Nitro` which would then use `$fetch`.

let me add this tonight : )

**Review by amritk — APPROVED** [review]
Lets go! and add `$fetch` after

**Comment by pi0** [comment]
Only `$fetch` for Nitro can be safely documented without an import. `ofetch` requires a clear import i think we are doing for undici but not ofetch... (web needs import from `esm.sh/ofetch` too also)

image

image

**Pull request merged**